src/utils/satisfyVersion.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+/**
+ * Checks if given version satisfy requirements (newer than)
+ * Versions are expected to have format like '29.1.4.3'
+ * @param version given version
+ * @param requirement version to compare against
+ */
+function satisfyVersion(version: string, requirement: string): boolean {
+	const versionMap = version.split('.').map(Number)
+	const requirementMap = requirement.split('.').map(Number)
```

Comment:
This won't work with pre-releases

Reply:
We can use [semver](https://www.npmjs.com/package/semver) package

Reply:
I changed it to rely on `version` and not `versionstring`. This comes from version.php, so should always be number tuple?
```php
$OC_Version = [31, 0, 0, 3];

// The human-readable string
$OC_VersionString = '31.0.0 dev';
```

Reply:
for the server version yes, for talk version no

Reply:
For talk we have capabilities to count such cases.
And I don't feel like adding semver to the prod dependencies, given that it's check is relevant only for web 29 and desktop client